### PR TITLE
Update to GNOME runtime 3.32 and Gnote 3.32.1

### PR DESCRIPTION
**Pull request by wjt**
~~This also updates the gtkmm, etc. dependencies by pulling them in from shared-modules, which depends on my PR at https://github.com/flathub/shared-modules/pull/54.~~

TODO:

- [x] Update to Gnote 3.32.1, released since I originally opened this PR
- [x] Inline gtkmm etc. updates since the PR above was not merged
- [x] Backport OARS data from upstream appdata file [MR](https://gitlab.gnome.org/GNOME/gnote/merge_requests/1)
- [x] Improve appdata (including `<releases>`)
- [x] [Submit improved appdata upstream](https://gitlab.gnome.org/GNOME/gnote/merge_requests/4)

### CODE REVIEW

**Comment by barthalion** [comment]
bot, build

**Comment by wjt** [comment]
Oh right, and that shared-modules PR was closed :( OK, some updates needed.

File: 0001-Add-OARS-and-update-appdata.patch [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
++ <project_license>GPL-3.0-only and GFDL-1.1-only</project_license>
+  <_name>Gnote</_name>
+  <_summary>A simple note-taking application</_summary>
+  <description>
+@@ -15,6 +16,38 @@
+   <screenshot type="default" width="624" height="351">https://wiki.gnome.org/Apps/Gnote?action=AttachFile&amp;do=get&amp;target=gnote_main_window.png</screenshot>
+  </screenshots>
+  <url type="homepage">https://wiki.gnome.org/Gnote</url>
++ <url type="bugtracker">https://gitlab.gnome.org/GNOME/gnote/issues</url>
++ <url type="donation">http://www.gnome.org/friends/</url>
```
Comment (TingPing) [inline 0001-Add-OARS-and-update-appdata.patch]: Probably best to not add donation fields to downstream files.

File: 0001-Add-OARS-and-update-appdata.patch [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
++ <url type="translate">https://wiki.gnome.org/TranslationProject</url>
+  <updatecontact>aurisc4_at_gmail.com</updatecontact>
++ <content_rating type="oars-1.1">
```
Comment (TingPing) [inline 0001-Add-OARS-and-update-appdata.patch]: This should be an empty list.

File: 0001-Add-OARS-and-update-appdata.patch (hunk 1 above)
Comment (wjt) [reply 0001-Add-OARS-and-update-appdata.patch]: [I didn't.](https://gitlab.gnome.org/GNOME/gnote/merge_requests/1)

File: 0001-Add-OARS-and-update-appdata.patch (hunk 2 above)
Comment (wjt) [reply 0001-Add-OARS-and-update-appdata.patch]: OK, I've added another patch to make that change that I'll submit upstream if this works OK.

**Comment by wjt** [comment]
It runs, the generated appstream looks OK, and I think I'm done for today :)

**Pull request merged**